### PR TITLE
Release artifacts

### PR DESCRIPTION
**Pull request by jburel**
This PR creates and uploads the release artifacts when a tag is pushed
4b1f6859f76cae9feb4865b66c19cb487699b903 is needed since the reference ``compile/runtime.classpath`` is no longer found when ant 1.10 is used (that's the version installed via GHA). Open to alternative is someone knows one

Tested via https://github.com/jburel/bioformats/releases/tag/v6.11.1-m1

### CODE REVIEW

**Review by sbesson — COMMENTED** [review]
A few additional questions in terms of the proposed changes:

1- I am going back and forth about injecting the `resolver:resolve` step in `components/bundle/bioformats_package.xml` as proposed here vs centralizing all the targets involving some dependency resolution to `ant/java.xml` 
2a- should the `docs` target be built as part of every run to catch future regressions?
2b- https://github.com/ome/bioformats/pull/3912 also modifies the logic of the Ant action to upload a set of artifacts for consumption. This addition could also be constructed a tag-specific step of the `ant` workflow assuming the relevant artifacts are all built

No blockers but possibly a few decisions that might be worth discussing as part of the @ome/formats weekly meeting.

**Comment by jburel** [comment]
> 1- I am going back and forth about injecting the `resolver:resolve` step in `components/bundle/bioformats_package.xml` as proposed here vs centralizing all the targets involving some dependency resolution to `ant/java.xml` 
I have tried various things to no success i.e. keeping it in `ant/java.xml` 


 >2a- should the `docs` target be built as part of every run to catch future regressions? 
This is an important part of the release process and it does not take long so I think it should be included by default.


> 2b- #3912 also modifies the logic of the Ant action to upload a set of artifacts for consumption. This addition could also be constructed a tag-specific step of the `ant` workflow assuming the relevant artifacts are all built

It could be. I follow the same pattern that we introduced in openmicroscopy mainly to  keep things clearer.
Convoluted workflows can be difficult sometimes to understand and adjust.
> 
> No blockers but possibly a few decisions that might be worth discussing as part of the @ome/formats weekly meeting.

File: components/bundles/bioformats_package/build.xml [hunk 1]
```diff
@@ -6,15 +6,26 @@ Download Apache Ant from http://ant.apache.org/.
...
   <property name="root.dir" location="../../.."/>
-  <import file="${root.dir}/ant/java.xml"/>
+  <import file="${root.dir}/ant/java.xml" />
   <property file="build.properties"/>
 
+
   <target name="docs" depends="javadoc-properties"
     description="generate the Javadocs for most components">
     <echo>----------=========== Javadocs ===========----------</echo>
+    <path id="compile.classpath"/>
```
Comment (sbesson) [inline components/bundles/bioformats_package/build.xml:19]: As mentioned in a previous version of this commit, this is equivalent to the approach I took in https://github.com/ome/bioformats/pull/3923/commits/bf8076ec08fffc2a5a2deedd6e6240a33ecc11a8#diff-fdeb882f159787e553f8d2b7f340d498880a8a54e60ae5dee163671dc9009018R203-R211 to fix the bundle generation target.

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [BIOFORMATS-push#359](https://snoopycrimecop-ci.openmicroscopy.org/jenkins/job/BIOFORMATS-push/359/). See the [console output](https://snoopycrimecop-ci.openmicroscopy.org/jenkins/job/BIOFORMATS-push/359/consoleText) for more details.
Possible conflicts:
  - PR #3815 sbesson 'Java 17 support'
    - .github/workflows/ant.yml
  - PR #3912 melissalinkert 'Upload one set of artifacts from the GitHub Actions ant build'
    - .github/workflows/ant.yml
  - PR #3933 jburel 'upgrade setup-java action. Current version uses set-output'
    - .github/workflows/ant.yml

--conflicts

**Review by dgault — APPROVED** [review]
Overall the changes look good, agreed that `ant docs` should indeed be part of the process. The artifacts generated by the test in https://github.com/jburel/bioformats/releases/tag/v6.11.1-m1 all look to be correct.

**Pull request merged**